src/libAtomVM/externalterm.c

```diff
@@ -183,14 +180,14 @@ term externalterm_to_binary(Context *ctx, term t)
     // Ensure enough free space in heap for binary
     //
     int size_in_terms = term_binary_data_size_in_terms(len);
-    if (UNLIKELY(memory_ensure_free(ctx, size_in_terms + 1) != MEMORY_GC_OK)) {
+    if (UNLIKELY(memory_ensure_free_maybe_gc(ctx, size_in_terms + 1) != MEMORY_GC_OK)) {
         fprintf(stderr, "Unable to ensure %i free words in heap\n", size_in_terms);
         return term_invalid_term();
     }
     //
     // create and return the binary
     //
-    term binary = term_from_literal_binary((void *) buf, len, ctx);
+    term binary = term_from_literal_binary(ctx->global, &ctx->heap, (void *) buf, len);
```

Comment:
I have been trying to follow Davide's pattern of putting Context-related structures at the end of the list of parameters.  In general, I prefer the pattern you have here, but for consistency, I have been trying to follow Davide's pattern.

Reply:
Thanks. I did update all `term_*` functions to have `GlobalContext *` last. It seems that `Heap *` should be penultimate, being not the main input.